frontend/src/hooks/useTerminal.ts

```diff
@@ -131,10 +134,17 @@ export const useTerminal = (commands: Command[] = []) => {
     if (terminal.current && commands.length > 0) {
       // Start writing commands from the last command index
       for (let i = lastCommandIndex.current; i < commands.length; i += 1) {
-        const command = commands[i];
-        terminal.current?.writeln(parseTerminalOutput(command.content));
+        // eslint-disable-next-line prefer-const
+        let { content, type } = commands[i];
 
-        if (command.type === "output") {
+        secrets.forEach((secret) => {
+          const secretRegex = new RegExp(secret, "g");
+          content = content.replace(secretRegex, "*".repeat(10));
```

Comment:
**Suggestion:** Optimize the secret replacement logic for performance.

**Existing Code:**
```TypeScript
secrets.forEach((secret) => {
  const secretRegex = new RegExp(secret, "g");
  content = content.replace(secretRegex, "*".repeat(10));
});
```

**Improved Code:**
```TypeScript
const secretRegex = new RegExp(secrets.join('|'), "g");
content = content.replace(secretRegex, "*".repeat(10));
```

**Details:**
Use a single regular expression to replace all secrets in one pass, improving performance when multiple secrets are present.

Reply:
Nice suggestion. The only thing to consider is that `secrets` must have elements, otherwise it will replace the space between letters.